kratos/processes/replace_elements_and_condition_process.cpp

```diff
@@ -23,52 +23,59 @@ namespace Kratos {
 namespace {
 
 template <class TEntity, class TEntityContainer>
-void ReplaceEntities(const TEntity& rReferenceEntity,
-                     TEntityContainer& rEntityContainer)
+void ReplaceEntities(
+    const TEntity& rReferenceEntity,
+    TEntityContainer& rEntityContainer,
+    const std::vector<int>& rListOfIds)
 {
     #pragma omp parallel for
     for (int i=0; i<static_cast<int>(rEntityContainer.size()); ++i) {
         auto it_entity = rEntityContainer.begin() + i;
+        if (std::find(rListOfIds.begin(), rListOfIds.end(), it_entity->Id()) != rListOfIds.end()) {
```

Comment:
hm I think this is a killer for larger models
I thought there would be a solution without this but now I don't remember
I will think abt it

Reply:
It would be better to use `std::unordered_set` to store the ids and retrieve them fast base on hash base find of this container

Reply:
you are right, i missed this. This is easily solved by  using internally an unordered set.

Reply:
Done! That really solved the problem. For a 1M test it went from spending 4 minutes to 0.5 secs  :' )